packages/react/src/components/TextField/TextField.test.tsx

```diff
@@ -527,78 +483,86 @@ describe('TextField with error message', () => {
     );
     jest.runAllTimers();
     expect(validator).toHaveBeenCalledTimes(0);
-    assertErrorMessage(wrapper.getDOMNode(), false);
+    assertErrorMessage(container, false);
 
-    wrapper.setProps({ value: 'failValidationValue' });
+    rerender(
+      <TextField
+        validateOnFocusOut
+        value="failValidationValue"
+        onChange={noOp}
+        onGetErrorMessage={validator}
+        validateOnLoad={false}
+      />,
+    );
     jest.runAllTimers();
 
     expect(validator).toHaveBeenCalledTimes(0);
-    assertErrorMessage(wrapper.getDOMNode(), false);
+    assertErrorMessage(container, false);
   });
 
   it('should trigger validation only on focus', () => {
     const validator = jest.fn((value: string) => (value.length > 3 ? errorMessage : ''));
 
-    wrapper = mount(<TextField defaultValue="initial value" onGetErrorMessage={validator} validateOnFocusIn />);
+    const { container } = render(
+      <TextField defaultValue="initial value" onGetErrorMessage={validator} validateOnFocusIn />,
+    );
 
-    const input = wrapper.find('input');
-    input.simulate('input', mockEvent('invalid value'));
+    const input = getInput();
+    userEvent.type(input, 'invalid value', { skipClick: true });
```

Comment:
What was the reason for skipClick here?

Reply:
The `validator` callback gets called when `Textfield` initially renders with the `validateOnFocusIn` prop passed, . So I added the `skipClick:true` because without it, the number of times the `validator` gets called would be two instead of one.

Reply:
In that case maybe it's better to remove skipClick and update the expected number of times the validator is called?